docs/core/entity/calendar.md

```diff
@@ -48,8 +48,8 @@ A calendar entity can return events that occur during a particular time range. S
 - The `end_date` is the upper bound and applied to the event's `start` (exclusive).
 - Recurring events should be flattened and returned as individual `CalendarEvent`.
 
-The start and end datetime will always have a timezone, which the implementation should be use for
-determing the order of events e.g. the start of an all day event for comparison with other events.
+The start and end datetime will always have a timezone. The implementation should use the timezone
+in the start datetime to determine the ordering of events and start time of an all day event.
```

Comment:
An alternative is integrations are instructed to always use `dt_util.DEFAULT_TIME_ZONE` as [local calendar](https://github.com/home-assistant/core/blob/ec33f6fe783fbbfe7c658fe92b2e0f6dbc16a18e/homeassistant/components/local_calendar/calendar.py#L90) and `dt_util.as_local` or [google](https://github.com/home-assistant/core/blob/ec33f6fe783fbbfe7c658fe92b2e0f6dbc16a18e/homeassistant/components/local_calendar/calendar.py#L83) do today.

Rationale: My thinking was that we decided the gcal library should just accept the timezone as its start/date time argument and I think the same thing applies here too.

Reply:
I think it's fine like it is now.